src/otx/cli/utils/hydra.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Functions related to Hydra."""
+
+from __future__ import annotations
+
+import copy
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from hydra.core.hydra_config import HydraConfig
+from hydra.core.utils import _save_config, configure_log
+
+if TYPE_CHECKING:
+    from omegaconf import DictConfig
+
+
+def configure_hydra_outputs(cfg: DictConfig) -> None:
+    """Configures the outputs for Hydra.
+
+    Copied some of the log and output configuration code from run_job fucntion of 'hydra.core.utils'.
+
+    Args:
+        cfg (DictConfig): The configuration dictionary.
+
+    Returns:
+        None
+    """
+    output_dir = Path(cfg.base.output_dir).resolve()
+    cfg.hydra.job_logging.handlers.file.filename = output_dir / f"{cfg.hydra.job.name}.log"
+    cfg.hydra.runtime.output_dir = output_dir
+    HydraConfig.instance().set_config(cfg)
```

Comment:
Please do not adjust the configuration and make it mutable at the runtime by the code which cannot easily be traced anymore just like OTX v1.
Did you try the alternative https://hydra.cc/docs/configure_hydra/workdir/?

Reply:
When I replaced hydra.main with hydra.compose, I realized that there was a difference in this function (https://github.com/facebookresearch/hydra/blob/809718cdcd64f9cd930d26dea69f2660a6ffa833/hydra/core/utils.py#L99) and I wrote this.
hydra.main = hydra.compose + run_job()
The problem is that this function called run_job modifies the output paths in the https://github.com/facebookresearch/hydra/blob/809718cdcd64f9cd930d26dea69f2660a6ffa833/hydra/core/utils.py#L116 part.

If you know if there is another way, can you give me a suggestion? I'll look for more, too.

Reply:
As far as I understand, you are trying to set `hydra.runtime.output_dir` from `base.output_dir`. However, `base.output_dir` is already the interpolated variable from `hydra.runtime.output_dir`.
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/cbb0cedb-7199-4e34-8e4d-80aa667d3c58)

If you want to change `hydra.runtime.output_dir`. I think those are possible:
1) Directly override from the CLI `hydra.runtime.output_dir=some_arbitrary_path`. This will change `base.output_dir` as well you know.
2) Let `hydra.runtime.output_dir` be interpolated by `base.output_dir` and change `base.output_dir` from the CLI.
```yaml
hydra:
  runtime:
    output_dir: ${base.output_dir}
```

Reply:
Thank you for the good suggestion. I modified it like this, so could you check it out?
https://github.com/openvinotoolkit/training_extensions/pull/2679/commits/ff21f338d1c39921a7385a23dffe814f6de76a8c
The above modifications still use the commands below and work well.
`otx train +recipe=detection/atss_r50_fpn base.data_dir=tests/assets/car_tree_bug model.otx_model.config.bbox_head.num_classes=3 trainer.max_epochs=50 trainer.check_val_every_n_epoch=10 trainer=gpu base.work_dir=outputs/test_work_dir base.output_dir=outputs/test_output_dir`

Reply:
Following this conversation, are those lines needed as well?
https://github.com/openvinotoolkit/training_extensions/blob/13a374d003bb487671c9cc3f3e189a37a6186d33/src/otx/cli/utils/hydra.py#L38-L45

Reply:
> Following this conversation, are those lines needed as well?
> 
> https://github.com/openvinotoolkit/training_extensions/blob/13a374d003bb487671c9cc3f3e189a37a6186d33/src/otx/cli/utils/hydra.py#L38-L45

If we don't need to keep the `outputs/.hydra` folder, we may not need it. But for now, I imported that code to preserve the output of the existing script as much as possible.

Reply:
You already changed `hydra.runtime.output_dir` as what you want. What is the reason that this function to save them again?
![image](https://github.com/openvinotoolkit/training_extensions/assets/26541465/92f0cfb1-6cb2-4fa7-878e-2f88d53cdc97)

Reply:
If those lines disappear, they don't create `.hydra` and files.

Reply:
We dropped `hydra.main`. I got what you mean.